### ISSUE TITLE
`pipenv install` should do minimal changes to `Pipfile.lock`

### ISSUE THREAD

**Issue by vporton** [pypa/pipenv#6070]
`pipenv install` should do minimal changes to `Pipfile.lock`. Now it updates versions of packages unrelated to a package specified in `pipenv install` wreaking havoc in package versions.

`npm` and `yarn` do the correct thing on this (I assume).

**Comment by vporton** [comment]
The best workaround would be a tool to migrate `.lock` from `pipenv` to `poetry` format.

**Comment by matteius** [comment]
What you first were asking for exists -- it is the `pipenv upgrade` command.

**Comment by vporton** [comment]
@matteius You misunderstand. I want `pipenv` **not** to run upgrade, when I for example add a new package.

`pipenv` `Pipfile.lock` is useless, because it unlike `poetry`, `npm`, `yarn`, `cargo` does not **lock** files. The purpose of `.lock` file is record installed file versions, so that on a different machine, for example, or after adding a new dependency the same versions would be installed, so minimizing the risk of version incompatibility. That is the meaning of the word **lock**. `pipenv` on the other hand does not respect locked file versions but when I add a new package (or remove or something), screws all "locked" data.

In other words, `pipenv install` should **not** behave, as if also `pipenv upgrade` were run.

**Comment by matteius** [comment]
Actually @vporton -- I think you misunderstand:   `upgrade` does not affect every package in the lock file, only ones that are direct sub-dependencies of the thing you are upgrading.  If you think there should be a way of updating individual dependencies without including their requires sub-dependencies, well that is just wrong.   `npm` `yarn` and others also manage the sub-dependencies required by a package.

**Comment by vlagorsse** [comment]
I'm actually mind-blowed by the answer, this is a problem I was actually blocked now. There was also this thread other there: https://github.com/pypa/pipenv/issues/5992

If I may do this feedback, I would encourage to document this behaviour in the documentation, this could help a lot of people.

**Comment by matteius** [comment]
I am closing this out -- sounds like you have a path forward; we have separate issues tracking  changing the install path to do more like `pipenv sync` and to use the `upgrade` paths when adding packages, which is slated for a major release.
